### PR TITLE
[SCRIPT] Add flag to use ccache

### PR DESCRIPTION
**Pull request by jopperm**
Add a flag to enable the use of `ccache` in the LLVM and Triton builds.

### CODE REVIEW

**Comment by jopperm** [comment]
@whitneywhtsang @sommerlukas @victor-eds, could you have a look please?

File: scripts/compile-triton.sh [hunk 1]
```diff
@@ -138,7 +150,8 @@ build_llvm() {
     -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
     -DCMAKE_INSTALL_PREFIX=$PACKAGES_DIR/llvm \
     -DCMAKE_C_COMPILER=$C_COMPILER \
-    -DCMAKE_CXX_COMPILER=$CXX_COMPILER ..
+    -DCMAKE_CXX_COMPILER=$CXX_COMPILER \
+    $ADDITIONAL_FLAGS
```
Comment (sommerlukas) [inline scripts/compile-triton.sh:154]: Would this immediately override the `CMAKE_CXX_COMPILER` set in the previous line?

Comment (jopperm) [reply scripts/compile-triton.sh:154]: I don't think so, but how do you mean? Note that we're setting the `...COMPILER_LAUNCHER=` variables via `$ADDITIONAL_FLAGS`, not `...COMPILER=`.

Comment (sommerlukas) [reply scripts/compile-triton.sh:154]: Ah, I missed that, thanks.

**Review by sommerlukas — APPROVED** [review]

**Pull request merged**